src/lib/convert.ts

```diff
@@ -102,14 +144,20 @@ class Transpiler {
               if (statement.program.blockParams) {
                 // {{#each foo as |key, value|}
                 const blockParams = statement.program.blockParams;
-                this.buffer.push(`{% for ${blockParams[1]}, ${blockParams[0]} in ${condition} %}`);
+                // {{#each foo as |k, v|} => has 2 variable in the same context, k and v
+                this.contexts.push([blockParams[0], blockParams[1]]);
+                this.buffer.push(
+                  `{% for ${blockParams[1]}, ${blockParams[0]} in ${condition}.items %}`,
+                );
               } else {
                 // {{#each foo}}
                 scope = `${condition}_i`;
+                this.contexts.push(scope);
```

Comment:
We should never be mutating the current context, but instead clone to create a new context.

Maybe push can just return a clone with the new item added.